src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef PRIMAL_EVAL_INTEGRAL_IMPL_HPP_
+#define PRIMAL_EVAL_INTEGRAL_IMPL_HPP_
+
+// Axom includes
+#include "axom/config.hpp"  // for compile-time configuration options
+#include "axom/primal.hpp"
+
+// MFEM includes
+#ifdef AXOM_USE_MFEM
+  #include "mfem.hpp"
+#else
+  #error "Primal's integral evaluation functions require mfem library."
+#endif
+
+// C++ includes
+#include <cmath>
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+/*!
+ * \brief Computes the tangent vector to a Bezier curve.
+ * 
+ * Uses the de Casteljau algorithm. Functionality will eventually
+ * be supplanted by Bezier curve class method.
+ * 
+ * \param [in] c the Bezier curve
+ * \param [in] t on [0, 1] the parameter value on which to evaluate the derivative
+ * \return the tangent vector
+ */
+template <typename T>
+primal::Vector<double, 2> temp_dt(const primal::BezierCurve<double, 2>& c, T t)
+{
+  Vector<double, 2> val;
+
+  const int ord = c.getOrder();
+  std::vector<T> dCarray(ord + 1);
+
+  // Run de Casteljau algorithm on each dimension
+  for(int i = 0; i < 2; ++i)
+  {
+    for(int p = 0; p <= ord; ++p)
+    {
+      dCarray[p] = c[p][i];
+    }
+
+    // stop one step early and take difference of last two values
+    for(int p = 1; p <= ord - 1; ++p)
+    {
+      const int end = ord - p;
+      for(int k = 0; k <= end; ++k)
+      {
+        dCarray[k] = (1 - t) * dCarray[k] + t * dCarray[k + 1];
+      }
+    }
+    val[i] = ord * (dCarray[1] - dCarray[0]);
+  }
+
+  return val;
+}
+
+/*!
+ * \brief Evaluate a scalar field line integral on a single Bezier curve.
+ *
+ * Evaluate the scalar field line integral with MFEM integration rule
+ *
+ * \param [in] c the Bezier curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] quad the mfem integration rule containing nodes and weights
+ * \return the value of the integral
+ */
+inline double evaluate_line_integral_component(
+  const primal::BezierCurve<double, 2>& c,
+  std::function<double(Point2D)> integrand,
+  const mfem::IntegrationRule* quad)
+{
+  // Store intermediate values
+  Point2D x_q;
+  Vector2D dx_q;
+
+  // Store/compute quadrature result
+  double full_quadrature = 0.0;
+  for(int q = 0; q < quad->GetNPoints(); q++)
+  {
+    // Get intermediate quadrature point
+    //  at which to evaluate tangent vector
+    x_q = c.evaluate(quad->IntPoint(q).x);
+    dx_q = detail::temp_dt(c, quad->IntPoint(q).x);
+
+    full_quadrature += quad->IntPoint(q).weight * integrand(x_q) * dx_q.norm();
```

Comment:
Minor (here and elsewhere): 
The compiler might be better able to optimize with temporaries for `x_q` and `dx_q`.
E.g.
```suggestion
  // Store/compute quadrature result
  double full_quadrature = 0.0;
  for(int q = 0; q < quad->GetNPoints(); q++)
  {
    // Get intermediate quadrature point
    //  at which to evaluate tangent vector
    auto x_q = c.evaluate(quad->IntPoint(q).x);
    auto dx_q = detail::temp_dt(c, quad->IntPoint(q).x);

    full_quadrature += quad->IntPoint(q).weight * integrand(x_q) * dx_q.norm();
```